compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -474,4 +487,58 @@ private void checkUniqueness(BLangConstant constant) {
             }
         }
     }
+
+    private void updateSymbolType(BLangConstant constant) {
+        if (constant.symbol.kind == SymbolKind.CONSTANT && constant.symbol.type.getKind() != TypeKind.FINITE &&
```

Comment:
Shall we extract `constant.symbol` out to a variable?

Reply:
Sure. Will update.